examples/social/README.md

```diff
@@ -0,0 +1,173 @@
+-   `deserializeBlock`
+-   `serializeState`
+-   `deserializeState`
+-   `serializeUpdate`
+-   `deserializeUpdate`
+-   `setCalculatedState`
+-   `getCalculatedState`
+-   `hashCalculatedState`
+-   `routes`
+
+For a more detailed understanding, please refer to the [complete documentation](https://docs.constellationnetwork.io/sdk/frameworks/currency/data-api) on the Data API.
+
+### Lifecycle Functions
+
+#### -> `validateUpdate`
+
+-   This method initiates the initial validation of updates on the L1 layer. Due to a lack of contextual information (state), its validation capabilities are constrained. Any errors arising from this method result in a 500 response from the `/data` POST endpoint.
+
+#### -> `validateData`
+
+-   This method validates data on the L0 layer, with access to contextual information, including the current state. In this example, we ensure that the provided address matches the one that signed the message. Additionally, we verify the most recent update timestamp to prevent the acceptance of outdated or duplicated data.
+
+#### -> `combine`
+
+-   This method takes validated data and the prior state, combining them to produce the new state. In this instance, we update the post information or subscriptions in the state based on the validated update.
+
+#### -> `dataEncoder` and `dataDecoder`
+
+-   These are the encoder/decoder components used for incoming updates.
+
+#### -> `calculatedStateEncoder`
+
+-   This encoder is employed for the calculatedState.
+
+#### -> `signedDataEntityDecoder`
+
+-   This function handles the parsing of request body formats (JSON, string, xml) into a `Signed[Update]` class.
+
+#### -> `serializeBlock` and `deserializeBlock`
+
+-   The serialize function accepts the block object and converts it into a byte array for storage within the snapshot. The deserialize function is responsible for deserializing into Blocks.
+
+#### -> `serializeState` and `deserializeState`
+
+-   The serialize function accepts the state object and converts it into a byte array for storage within the snapshot. The deserialize function is responsible for deserializing into State.
+
+#### -> `serializeUpdate` and `deserializeUpdate`
+
+-   The serialize function accepts the update object and converts it into a byte array for storage within the snapshot. The deserialize function is responsible for deserializing into Updates.
+
+#### -> `setCalculatedState`
+
+-   This function sets the calculatedState. You can store this as a variable in memory or use external services such as databases. In this example, we use in-memory storage.
+
+#### -> `getCalculatedState`
+
+-   This function retrieves the calculated state.
+
+#### -> `hashCalculatedState`
+
+-   This function creates a hash of the calculatedState to be validated when rebuilding this state, in case of restarting the metagraph.
+
+#### -> `routes`
+
+Customizes routes for our application.
+
+In this example, the following endpoints are implemented:
+
```

Comment:
Let's add some basic resource endpoints to make this easier to explore:
 - /users: endpoint that lists users that have posts. 
 - /posts: endpoint that returns all posts

Reply:
Updated